juju/model.py

```diff
@@ -301,6 +301,12 @@ def entity_type(self):
         'application' or 'unit', etc.
 
         """
+        # Allow the overriding of entity names from the type instead of from
+        # the class name. Useful because Model and ModelInfo clash and we really
+        # want ModelInfo to be called Model.
+        if hasattr(self.__class__, "entity_name") and callable(self.__class__.entity_name):
+            return self.__class__.entity_name()
+
```

Comment:
I'd rather see this called "type_override" or something similar. I like the principle, but the name is confusing.

Reply:
Sure, I don't mind that.